File: src/main/java/com/topably/assets/portfolios/domain/PortfolioHoldingView.java
package com.topably.assets.portfolios.domain;

import com.topably.assets.core.domain.Ticker;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.math.BigInteger;

@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PortfolioHoldingView {

    private Long id;
    private Long instrumentId;
    private String instrumentType;
    private Ticker identifier;
    private BigInteger quantity;
    private BigDecimal price;
    private String currencySymbol;
    private BigDecimal pctOfPortfolio;
    private BigDecimal marketValue;

    public BigDecimal getTotal() {
        return getPrice().multiply(new BigDecimal(getQuantity()));
    }
}

File: client/src/app/features/holdings/types/portfolio-holding.view.ts
import { Ticker } from '../../../core/types/ticker';

export interface PortfolioHoldingView {
  id: number;
  instrumentId: number;
  instrumentType: any;
  identifier: Ticker;
  quantity: number;
  price: number;
  currencySymbol: string;
  pctOfPortfolio: number;
  marketValue: number;
  yieldOnCost: number;
}
